api/rbac/v1alpha1/annotations.go

```diff
@@ -46,23 +49,50 @@ func OIDCClaimNameFromAnnotationKey(key string) (string, bool) {
 	return strings.TrimPrefix(key, AnnotationKeyOIDCClaimNamePrefix), true
 }
 
-// OIDCClaimsFromAnnotationValue parses the value of an
-// rbac.kargo.akuity.io/claims annotation and returns the corresponding list of
-// formatted and sorted claims.
-func OIDCClaimsFromAnnotationValue(value string) ([]string, error) {
+// OIDCClaimsFromAnnotationValue parses the values of both the
+// rbac.kargo.akuity.io/claims and rbac.kargo.akuity.io/claim.<name> annotations
+// and consolidates them into a single map where the value of each key is sorted and deduped.
+func OIDCClaimsFromAnnotationValues(annotations map[string]string) (map[string][]string, error) {
 	claims := make(map[string][]string)
-	if err := json.Unmarshal([]byte(value), &claims); err != nil {
-		return nil, fmt.Errorf("unmarshaling OIDC claims from annotation value: %w", err)
+	// hydrate with new style claims
+	if _, ok := annotations[AnnotationKeyOIDCClaims]; ok {
+		if err := json.Unmarshal([]byte(annotations[AnnotationKeyOIDCClaims]), &claims); err != nil {
+			return nil, fmt.Errorf("unmarshaling OIDC claims from annotation value: %w", err)
+		}
+	}
+	// hydrate with old style claims
+	for name, values := range annotations {
+		if key, ok := OIDCClaimNameFromAnnotationKey(name); ok {
+			for v := range strings.SplitSeq(values, ",") {
+				claims[key] = append(claims[key], strings.TrimSpace(v))
+			}
+		}
+	}
+	for k, v := range claims {
+		slices.Sort(v)
+		claims[k] = slices.Compact(v)
+	}
+	return claims, nil
+}
+
+// SetOIDCClaimsAnnotations sets the rbac.kargo.akuity.io/claims annotation on
+// the given ServiceAccount to the JSON representation of the given claims map.
+// It also removes any existing rbac.kargo.akuity.io/claim.<name> annotations.
+func SetOIDCClaimsAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) error {
```

Comment:
Good call making a reusable function for this also.

Only suggestion is this is more accurate if it's `SetOIDCClaimsAnnotation()` (singular) and there's only one claims annotation in the new claims --> SA mapping scheme.

```suggestion
// SetOIDCClaimsAnnotation sets the rbac.kargo.akuity.io/claims annotation on
// the given ServiceAccount to the JSON representation of the given claims map.
// It also removes any existing rbac.kargo.akuity.io/claim.<name> annotations.
func SetOIDCClaimsAnnotation(sa *corev1.ServiceAccount, claims map[string][]string) error {
```

I am a bit iffy on this function also cleaning up the old annotations, as it feels like a distinctly different responsibility, but I cannot see any circumstance where we want to write out mappings using the new annotation _without_ also cleaning up the older annotations, so this definitely makes sense. Let's leave it be for now, but it's something we might come back to.